### PR TITLE
Add support for CSS color function notation to Led.RGB.color()

### PR DESCRIPTION
**Pull request by islemaster**
Adds support for CSS color function notation (e.g. `"rgb(255, 255, 0)"`) to the `color()` method on the Led.RGB prototype.  Based on the [CSS Color Module Level 4 Draft](https://drafts.csswg.org/css-color/#rgb-functions) and [the Color article at MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#rgb()).

You can now call `led.color()` with the following strings:

  - `"rgb(r, g, b)"` where `r`, `g`, and `b` are numbers in the range 0-255.
  - `"rgb(r, g, b, a)"` where `a` is in the range 0-1 and modifies the color as an intensity function using `RGB.ToScaledRGB()`
  - `"rgba(r, g, b)"` or `"rgba(r, g, b, a)"` because `rgba` is an alias of `rgb` as of the Level 4 Draft.
  - `"rgb(r,g,b)"` or `"rgb(r g b)"` because arguments can be separated by whitespace, commas, or both as of the Level 4 Draft.

### CODE REVIEW

**Comment by islemaster** [comment]
Hmm :frowning_face: [the appveyor failure](https://ci.appveyor.com/project/rwaldron/johnny-five/build/781/job/2a3mak1tj15wcult) looks unrelated to my changes.  Is there a problem on master?
```
Testing animation.js........F.............................
>> Animation -- Servo - additiveEasing
>> Error: 0 == true
>> at Object.exports.Animation -- Servo.additiveEasing (test\animation.js:286:10)
>> at Object.exports.Animation -- Servo.setUp (test\animation.js:97:5)
```

**Comment by islemaster** [comment]
Similar failure, [this time on a Travis container](https://travis-ci.org/rwaldron/johnny-five/jobs/221163583):
```
Testing animation.js.......F..............................
>> Animation -- Servo - keyframeEasing
>> Error: 0 == true
>> at Object.exports.Animation -- Servo.keyframeEasing (test/animation.js:263:10)
>> at Object.exports.Animation -- Servo.setUp (test/animation.js:97:5)
```
Perhaps the animation tests are flaky?

**Comment by rwaldron** [comment]
Whoa, wacky. I love this idea.

File: test/rgb.js [hunk 1]
```diff
@@ -229,6 +229,59 @@ exports["RGB"] = {
     }));
     this.write.reset();
 
+    // CSS functional notation
+    this.rgb.color("rgb(255, 100, 50)");
```
Comment (rwaldron) [inline test/rgb.js]: 👍

File: test/rgb.js [hunk 2]
```diff
@@ -229,6 +229,59 @@ exports["RGB"] = {
...
+      red: 255,
+      green: 100,
+      blue: 50
+    }));
+    this.write.reset();
+
+    // CSS functional notation with alpha
+    // Subtle bug: Alpha is translated to intensity using linear scaling
+    // when it should probably use logarithmic scaling.
+    this.rgb.color("rgba(255, 100, 50, 0.5)");
```
Comment (rwaldron) [inline test/rgb.js]: 👍

File: test/rgb.js [hunk 3]
```diff
@@ -229,6 +229,59 @@ exports["RGB"] = {
...
+    test.ok(this.write.calledOnce);
+    test.ok(this.write.calledWith({
+      red: 128,
+      green: 50,
+      blue: 25
+    }));
+    this.write.reset();
+
+    // CSS4 functional notation without commas
+    this.rgb.color("rgb(255 100 50)");
```
Comment (rwaldron) [inline test/rgb.js]: Damn, that's sharp

**Comment by rwaldron** [comment]
Notes: 

- The failures are unrelated timing issues. 
- Excellent tests!

**Pull request merged**